certify-service/src/test/java/io/mosip/certify/controller/VCIssuanceControllerTest.java

```diff
@@ -84,6 +85,15 @@ public void getIssuerMetadata_withQueryParam_thenPass() throws Exception {
         Mockito.verify(vcIssuanceService).getCredentialIssuerMetadata("v11");
     }
 
+    @Test
+    public void getIssuerMetadata_withInvalidQueryParam_thenPass() throws Exception {
+        Exception e = new InvalidRequestException(ErrorConstants.UNSUPPORTED_OPENID_VERSION);
+        Mockito.when(vcIssuanceService.getCredentialIssuerMetadata("v123")).thenThrow(e);
+        mockMvc.perform(get("/issuance/.well-known/openid-credential-issuer?version=v123"))
+                .andExpect(status().is4xxClientError());
+        Mockito.verify(vcIssuanceService).getCredentialIssuerMetadata("v123");
```

Comment:
do we need this line?

Reply:
Yes. It verifies if the getCredentialIssuerMetadata function is called with `v123` argument.

Reply:
I'm now split on this.


**Reasons for**:
- verifies that the mock was exactly called for this specific argument

**Reasons against**:
- technically the system under test is the Controller which if returns a 4XX error is a sufficient test case

What do you suggest here?